airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/utils.py

```diff
@@ -1,13 +1,58 @@
 #
 # Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 #
-
+import logging
+import time
 
 from airbyte_cdk.utils import AirbyteTracedException
 from airbyte_protocol.models import FailureType
 
+LOG_LEVEL = logging.getLevelName("INFO")
+LOGGER = logging.getLogger("airbyte")
+
 
 class AmazonConfigException(AirbyteTracedException):
     def __init__(self, **kwargs):
         failure_type: FailureType = FailureType.config_error
         super(AmazonConfigException, self).__init__(failure_type=failure_type, **kwargs)
+
+
+class ReportRateLimits:
+    def __init__(self, threshold: int, period_in_minutes: int):
+        self.threshold = threshold
+        self.period_in_minutes = period_in_minutes
+
+    @property
+    def wait_time_in_seconds(self):
+        return (self.period_in_minutes / self.threshold) * 60
+
+
+# https://github.com/airbytehq/alpha-beta-issues/issues/3717#issuecomment-2203717834
+STREAM_THRESHOLD_PERIOD = {
+    "GET_AMAZON_FULFILLED_SHIPMENTS_DATA_GENERAL": ReportRateLimits(threshold=12, period_in_minutes=480),
```

Comment:
my bad 🙄 , missed docs update.

I'm good with last 2 streams, but i don't like the idea of `sleeping` for 40 min just to make another try. We've made this option for our GitHub connector, but the only reason was because we were unable to finish the reading process and checkpoint.
In our current situation, I'd prefer to raise an exception rather than abuse our cloud instance.
@girarda, WDYT?

Reply:
I'm fine with commenting this line:
`"GET_AMAZON_FULFILLED_SHIPMENTS_DATA_GENERAL": ReportRateLimits(threshold=12, period_in_minutes=480),`
Let's just keep it as documentation.

Reply:
commented for GET_AMAZON_FULFILLED_SHIPMENTS_DATA_GENERAL